deploy/kubernetes/examples/demo-storageclass-config-secret.yaml

```diff
@@ -0,0 +1,20 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: demo-storageclass
+provisioner: block.csi.ibm.com
+volumeBindingMode: WaitForFirstConsumer  # Optional. Required for Topology Awareness.
+parameters:
+  # non secret related parameters can be provided in by_system_id parameter for each system or by parameter name as default.
+
+  by_system_id: '{"demo-system-id-1":{"pool":"demo-pool-1","SpaceEfficiency":"thin","volume_name_prefix":"demo-prefix-1"},

```

Comment:
wouldn't it be more realistic to use a space efficiency here that is different than the one below?

Reply:
it's different from `demo-system-id-2` that has the default.

Reply:
but if both systems eventually get 'thin', then is it really different?

Reply:
The second will get thick/none (that is the default..)

Reply:
oh, then I misunderstood.
I thought it would get the value from [here](https://github.com/IBM/ibm-block-csi-driver/blob/6416b2e52d8e4465348fad54ab79664f0bd5b856/deploy/kubernetes/examples/demo-storageclass-config-secret.yaml#L13).
it would be nice if this nuance is clear from the yaml comment

Reply:
oh, it will, I thought you only talked about  `demo-system-id-1` and `demo-system-id-2` as if there were no other optional parameters.
anyway, there is no other `"cross-storage" parameter to put in the second `SpaceEfficiency` parameter.

Reply:
is cross-storage more important than for the example to make sense?
I thought the prefix value showed that (unfortunately) cross-storage is not so significant to us

Reply:
there is a difference between "out-of-the-box" and "sense". "demo-prefix" is self-explanatory as it's a free string. `SpaceEfficiency` is not a free string parameter (and therefore we are not putting there demo-space-efficiency value) so it should contain one of the optional parameters that the user can put. no ds8k user can put `deduplication` so it's not a valid parameter to put there.

Reply:
free string or not, I think it is more important for the example to make sense than to work on both systems.
@zingero what do you think?

Reply:
it's not just "free string", to a non-SW user `deduplication` will not make any sense

Reply:
It's an example. It doesn't have to work out of the box or have different values in the `by_system_id`/the parameter below.
From my point of view, we can put a dummy value like "demo-space-efficiency" or an actual valid value that applies to all of our storage types (svc, a9k, ds8k). Since a9k doesn't support any space efficiency value, I suggest not using an actual value, but "demo-space-efficiency". This will make the user pick a specific SE that works with their storage type.

Reply:
@oriyarde what do you think?

Reply:
I think we each hold a different opinion

Reply:
perhaps the word "examples" is confusing.
the way I see it, ever since we moved to the "demo" approach, the goal was to provide a set of configurations that work with minimal change, to demonstrate the product flow

Reply:
@ArbelNathan what do you think?

Reply:
that we can use a third opinion

Reply:
IMO, the example yamls do not have to fit for each of the storages, as long as the documentation shows which `SpaceEfficiency` is supported for each of the storages

Reply:
sounds great

Reply:
> that we can use a third opinion

fourth*

Reply:
Thanks for another notification.

Reply:
np

Reply:
done